scripts/resolve_container_path.py

```diff
@@ -19,24 +27,44 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo_path = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo_path) as f:
+            log.info('Found %s', mountinfo_path)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…/…
+            # dc61d9… is the container ID
+            for line in f:
+                assert line.endswith('\n'), line
+                parts = line.strip().split(' ')
```

Comment:
This removes spaces appearing before the `\n` as well. The only you thing you know at this point is that there is `\n` at the end of the string, and that's what you should remove: the last character.


```suggestion
                parts = line[:-1].split(' ')
```